atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TimestampLockAwareTransaction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import java.util.function.Consumer;
+
+public interface TimestampLockAwareTransaction {
```

Comment:
How do I get access to this? doesn't this need to extend Transaction or something? Transaction is what everyone will get? or do you want to cast in AEQ?

Reply:
cast in AEQ

Reply:
Meh, but let's start with that